src/Symfony/Component/Yaml/Parser.php

```diff
@@ -286,7 +286,11 @@ public function parse($value, $exceptionOnInvalidType = false, $objectSupport =
             }
 
             if ($isRef) {
-                $this->refs[$isRef] = end($data);
+                if (!isset($key)) {
```

Comment:
yes I had to read the parser a few times to be confident it was doing what I wanted. I'll take a look at your other solution